### PR TITLE
feat(models): enable Qwen3.5 text-only (Qwen3_5ForCausalLM) — IsHybrid, SupportsMRoPE, VL weight remapping

### PR DESCRIPTION
**Pull request by groxaxo**
## Summary

This PR narrows Qwen3.5 text-only support to **config-driven compatibility** plus the **native runtime fixes** needed once the model resolves to the text-only causal LM path.

Specifically, it:
- registers `qwen3_5_text` / `qwen3_5_moe_text` so vLLM can parse text-only checkpoints that publish those `model_type` values
- remaps text-only Qwen3.5 configs that still advertise the VL `...ForConditionalGeneration` architectures onto the native `Qwen3_5ForCausalLM` / `Qwen3_5MoeForCausalLM` implementations
- keeps the runtime fixes required by those native text-only models in `vllm/model_executor/models/qwen3_5.py`
  - `IsHybrid` so hybrid attention + GatedDeltaNet cache sizing is computed correctly
  - `SupportsMRoPE` for inherited `mrope_section` configs
  - `WeightsMapper` plus ignored visual prefixes so VL-derived checkpoints can load the text-only LM weights cleanly
- adds **local config-based regression coverage** instead of depending on public unofficial text-only checkpoints for registry coverage

## Why this shape?

Some quantized or fine-tuned Qwen3.5 text-only checkpoints surface `model_type = qwen3_5_text` or `qwen3_5_moe_text`, and may still carry conditional-generation architecture names inherited from the VL parent config.

The goal here is to normalize those configs onto the native causal LM path already implemented in vLLM, rather than treat them as separate first-class public architectures.

## Non-goals

- This PR does **not** depend on public Qwen-owned text-only checkpoints for CI coverage.
- Existing `Qwen3_5ForConditionalGeneration` / `Qwen3_5MoeForConditionalGeneration` behavior is unchanged.

## Testing

- `python -m py_compile` on touched files
- `pre-commit run --files ...`
- `pre-commit run --hook-stage manual mypy-3.10 --files ...`
- config parser / model-arch smoke test for `qwen3_5_text` and `qwen3_5_moe_text` remapping
- local runtime bring-up previously validated on a quantized Qwen3.5 text-only checkpoint (hybrid block-size selection, M-RoPE path, and VL-to-text weight remap)

### CODE REVIEW

**Comment by DorBernsohn** [comment]
Hi @groxaxo,

We're hitting all four of these issues when trying to run evaluation on a fine-tuned `Qwen3.5-27B` text-only checkpoint with `vLLM` + `transformers>=5.2.0`. Combined with the `layer_type_validation` fix from [[#37398]](https://github.com/vllm-project/vllm/pull/37398) (now on `main`), this PR would completely unblock us.

Is there anything blocking this from being merged? Happy to help with testing if needed.

**Comment by groxaxo** [comment]
Thanks for the pointer — I reworked the branch in that direction and pushed `432bac3`.

What changed in this follow-up:
- normalize Qwen3.5 text-only configs through the model-arch convertor instead of relying on public unofficial checkpoint coverage
- add the missing `qwen3_5_moe_text` config registration and model-config mapping for the native `Qwen3_5ForCausalLM` / `Qwen3_5MoeForCausalLM` paths
- replace the registry coverage entries that pointed at public text-only repos with local config-based coverage
- add a focused config test that exercises the `qwen3_5_text` / `qwen3_5_moe_text` architecture remap directly

I left the runtime fixes in `qwen3_5.py` intact (hybrid/M-RoPE/weight remap), and the new follow-up passed `pre-commit run --files ...` plus the manual `mypy-3.10` hook locally. I also smoke-tested the config parser + arch remap path with temporary configs to make sure the text-only model types now resolve to the native causal LM implementations.

If you want, I can also trim the PR description so it reflects the narrower config-remap approach more accurately.

File: tests/models/registry.py [hunk 1]
```diff
@@ -488,6 +488,14 @@ def check_available_online(
     "Qwen2MoeForCausalLM": _HfExamplesInfo("Qwen/Qwen1.5-MoE-A2.7B-Chat"),
     "Qwen3ForCausalLM": _HfExamplesInfo("Qwen/Qwen3-8B"),
     "Qwen3MoeForCausalLM": _HfExamplesInfo("Qwen/Qwen3-30B-A3B"),
+    "Qwen3_5ForCausalLM": _HfExamplesInfo(
+        "local/qwen3_5_text_config_example",
```
Comment (DarkLight1337) [inline tests/models/registry.py]: Do you have real official HF checkpoints that actually use this architecture? The issue mentioned in the comments I linked before is that the official Qwen checkpoints use `*ForConditionalGeneration` instead of `*ForCausalLM`, so why do we need to support this alternative name?

Comment (jefcoder) [reply tests/models/registry.py]: Hi @DarkLight1337 , many models fine-tuned from Qwen3.5 model has "Qwen3_5ForCausalLM" architecture.

For example aidonuts/punctured_vesicle_s1000_a063-v2 and grobb-takes-all/research-qwen-4b-v1 has Qwen3_5ForCausalLM instead of Qwen3_5ForConditionalGeneration.

Then how can I turn these models to Qwen3_5ForConditionalGeneration?

**Comment by Oxi84** [comment]
Thanks, which transformers version need to be installed with this one?

**Comment by Oxi84** [comment]
> Hi @groxaxo,
> 
> We're hitting all four of these issues when trying to run evaluation on a fine-tuned `Qwen3.5-27B` text-only checkpoint with `vLLM` + `transformers>=5.2.0`. Combined with the `layer_type_validation` fix from [[#37398]](https://github.com/vllm-project/vllm/pull/37398) (now on `main`), this PR would completely unblock us.
> 
> Is there anything blocking this from being merged? Happy to help with testing if needed.

Did you manage to fix this in the end?

File: tests/models/registry.py (hunk 1 above)
Comment (Oxi84) [reply tests/models/registry.py]: > Do you have real official HF checkpoints that actually use this architecture? The issue mentioned in the comments I linked before is that the official Qwen checkpoints use `*ForConditionalGeneration` instead of `*ForCausalLM`, so why do we need to support this alternative name?

Finetuned models such as bnb lora are always saved as Qwen3_5ForCausalLM, and when when i change the structure to multimodal it does not work -  produces half garbage.
